### ISSUE TITLE
Searchfox strings are too specific

### ISSUE THREAD

**Issue by wlach** [mozilla/glean-dictionary#360]
e.g. on this page:

https://dictionary.protosaur.dev/apps/fenix/metrics/startup_timeline_framework_start

The searchfox URL is this:

https://searchfox.org/mozilla-mobile/search?q=startupTimeline.frameworkStart|StartupTimeline.frameworkStart|startup_timeline.framework_start|startup_timeline::framework_start&path=fenix&regexp=true

This returns no results. However, a simpler query returns a bunch:

https://searchfox.org/mozilla-mobile/search?q=%28framework_start%7CframeworkStart%29&path=fenix&regexp=true

I think we should probably just simplify the string that we send to searchfox. Thoughts @mdboom @badboy? (and cc @Iinh)

**Comment by mdboom** [comment]
Yeah -- this is a tricky area.  The issue is that the implementer of the telemetry did this: 

```
import org.mozilla.fenix.GleanMetrics.StartupTimeline as Telemetry
```

I'd recommend against that, but we can't stop people from doing it.  Supporting that would require some sort of deeper understanding of the code.

The reason I'm hesitant about just searching only on the name (rather than `category.name`) we'll just have too many false positives, particular if the word is more generic.  Personally, I'd rather document somewhere "if you don't use `import ... as ...` you'll get searchfox support..."

But just my 2c

**Comment by wlach** [comment]
> The reason I'm hesitant about just searching only on the name (rather than `category.name`) we'll just have too many false positives, particular if the word is more generic. Personally, I'd rather document somewhere "if you don't use `import ... as ...` you'll get searchfox support..."

It looks like searchfox will return JSON if you set the right headers, e.g.:

`curl "https://searchfox.org/mozilla-mobile/search?q=startupTimeline.frameworkStart|StartupTimeline.frameworkStart|startup_timeline.framework_start|startup_timeline::framework_start&path=fenix&regexp=true" -H "Accept: application/json"`

It is possible we could do a quick check to see if there are results for the more complicated query, then fall back to the simplified one (maybe with a warning) if that doesn't work. It looks like we'd have to modify searchfox to provide the right CORS headers for this to work, but that's probably not too heavy a lift.

Or we could just document this in the UI somewhere. Probably best not to over-rotate on this issue for now (I don't know how widespread it is).

**Comment by mdboom** [comment]
> > The reason I'm hesitant about just searching only on the name (rather than `category.name`) we'll just have too many false positives, particular if the word is more generic. Personally, I'd rather document somewhere "if you don't use `import ... as ...` you'll get searchfox support..."
> 
> It looks like searchfox will return JSON if you set the right headers, e.g.:
> 
> `curl "https://searchfox.org/mozilla-mobile/search?q=startupTimeline.frameworkStart|StartupTimeline.frameworkStart|startup_timeline.framework_start|startup_timeline::framework_start&path=fenix&regexp=true" -H "Accept: application/json"`
> 
> It is possible we could do a quick check to see if there are results for the more complicated query, then fall back to the simplified one (maybe with a warning) if that doesn't work. It looks like we'd have to modify searchfox to provide the right CORS headers for this to work, but that's probably not too heavy a lift.

That's an interesting solution.  If this becomes a frequent problem in the future, that's perhaps worth tackling.

> Or we could just document this in the UI somewhere. Probably best not to over-rotate on this issue for now (I don't know how widespread it is).

It looks like there's only one instance of this in fenix and one in android-components.  (Which equates to more than one metric, as it affects an entire source file).  But still, rare enough I'd say just document the shortcoming and move on for now.

**Comment by wlach** [comment]
Another datapoint -- Searchfox also fails for many iOS metrics: https://dictionary.protosaur.dev/apps/firefox-ios-dev/metrics/app_opened_as_default_browser

It may be worthwhile doing an investigation on exactly how many searchfox URLs resolve to meaningful results (e.g. by using BeautifulSoup + curl).

**Comment by mdboom** [comment]
This is an interesting one -- there is a hit for this in the [github repo](https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Telemetry/TelemetryWrapper.swift#L484), but searchfox seems to [have a different revision of the file](https://searchfox.org/mozilla-mobile/source/firefox-ios/Client/Telemetry/TelemetryWrapper.swift).  Definitely worth investigation, but it might not be an issue on glean dictionary's end...

**Comment by wlach** [comment]
> This is an interesting one -- there is a hit for this in the [github repo](https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Telemetry/TelemetryWrapper.swift#L484), but searchfox seems to [have a different revision of the file](https://searchfox.org/mozilla-mobile/source/firefox-ios/Client/Telemetry/TelemetryWrapper.swift). Definitely worth investigation, but it might not be an issue on glean dictionary's end...

Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1690341
